packages/open-truss/src/pages/config-builder-page/config-builder-context.tsx

```diff
@@ -0,0 +1,88 @@
+import get from 'lodash/get'
+import set from 'lodash/set'
+import { createContext, useState } from 'react'
+import { type FrameType, type WorkflowSpec } from '../../configuration'
+import { type YamlType, parseYaml, stringifyYaml } from '../../utils/yaml'
+
+export const CONFIG_BASE = `
+workflow:
+  version: 1
+  frameWrapper: ConfigBuilderFrameWrapper
+  frames: []
+`.trim()
+export const INITIAL_FRAMES_PATH = 'workflow.frames'
+
+interface ConfigBuilder {
+  config: string
+  setConfig: (config: string) => void
+  framesPath: string
+  setFramesPath: (path: string) => void
+  addFrame: (frame: FrameType) => void
+  deleteFrame: (framePath?: string) => void
+}
+export const ConfigBuilderContext = createContext<ConfigBuilder>({
+  config: CONFIG_BASE,
+  setConfig: (_c: string) => null,
+  framesPath: INITIAL_FRAMES_PATH,
+  setFramesPath: (_c: string) => null,
+  addFrame: (_f: FrameType) => null,
+  deleteFrame: (_c?: string) => null,
+})
+
+export const Provider: React.FC<React.PropsWithChildren> = ({ children }) => {
+  const [config, setConfig] = useState<string>(CONFIG_BASE)
+  const [framesPath, setFramesPath] = useState<string>(INITIAL_FRAMES_PATH)
+
+  const addFrame = (frame: FrameType): void => {
+    const parsedConfig = parseYaml(config) as unknown as WorkflowSpec
```

Comment:
Do you think there is any utility to eventually make this parsed by zod? That would run validations and give us the type as opposed to us having to cast to `unknown` then `WorkflowSpec`. If we don't get much from this here though, then probably not worth the effort.

Reply:
Interesting! I think not yet because the config builder constructs the frame for you from the zod description. However, I was thinking we probably want to support editing the config directly by typing in the YAML and in that case doing this would be helpful